### PR TITLE
Issue #13321: Kill mutation for SummaryJavaDocCheck-1

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13321: Kill mutation for SummaryJavaDocCheck-1

-----

# Check :- 
https://checkstyle.org/checks/javadoc/summaryjavadoc.html#SummaryJavadoc

-----

# Mutation 
https://github.com/checkstyle/checkstyle/blob/3d8ed42ab60c0399550b3e7d54ca9dd767583101/config/pitest-suppressions/pitest-javadoc-suppressions.xml#L858-L865

-----

# Explaination

The removal of trim will effect the programme it will increase the number of times in loop.

There are total 3 usage of the `containsForbiddenFragment`

1) In `validateUntaggedSummary` method at 
https://github.com/checkstyle/checkstyle/blob/6073cd4194af4e673a5805fcfc6a2cddb0e4547d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L343-L345
Note:-  If this `condition` execute then violation thrown and violation message doesn't contains any relation with the text.

Know when `containsForbiddenFragment` will be execute then https://github.com/checkstyle/checkstyle/blob/6073cd4194af4e673a5805fcfc6a2cddb0e4547d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L603-L606 
javadocText will not be trimmed lets suppose if we have sentence `This is summary javadoc                .` 
then javaDoctext would become `This is summary javadoc                ` 

and as a return statment we call `trimExcessWhitespaces` to match `forbiddenSummaryFragment` and in trimExcessWhitespaces for loop will run as more time as we have extra spaces which is not been removed as we remove trim.
and the string will return with the extra space but it will not affect the output. 

And for the second and third usage the explaintaion is same as above 
https://github.com/checkstyle/checkstyle/blob/6073cd4194af4e673a5805fcfc6a2cddb0e4547d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L528-L530
and 
https://github.com/checkstyle/checkstyle/blob/6073cd4194af4e673a5805fcfc6a2cddb0e4547d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L511-L513

-----

# Regression :- 

Report-1 :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/5d08f12_2023202818/reports/diff/index.html

Report-2:- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/5d08f12_2023084141/reports/diff/index.html

-----

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/61eaa69fe98585b6ee1882ca082362bc/raw/c74195e574d09886a3406be17a9d46787da9545e/Summary.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/9600f179b602d4c971bdb0a050099005/raw/360a95ed7bb60d7a0956e531199d484c4d6f6617/test-projects.properties
Report label: Regression-2

### CODE REVIEW

**Comment by Kevin222004** [comment]
Github, generate report

**Review by romani — APPROVED** [review]
Ok, looks like extra trim

**Review by rdiachenko — APPROVED** [review]
lgtm

File: config/pitest-suppressions/pitest-javadoc-suppressions.xml [hunk 1]
```diff
@@ -855,14 +855,14 @@
...
-    <description>replaced call to java/lang/String::trim with receiver</description>
-    <lineContent>.matcher(firstSentence).replaceAll(&quot; &quot;).trim();</lineContent>
-  </mutation>
+
+
+
+
+
+
+
```
Comment (Vyom-Yadav) [inline config/pitest-suppressions/pitest-javadoc-suppressions.xml:864]: Please remove the empty lines.

**Comment by Vyom-Yadav** [comment]
@Kevin222004  I have found a regression test case with the current changes:

With the original code:

```java
@Test                                                                                         
public void testAsd() throws Exception {                                                      
    final String[] expected = {                                                               
            "16: " + getCheckMessage(MSG_SUMMARY_JAVADOC),                                    
    };                                                                                        
                                                                                              
    verifyWithInlineConfigParser(                                                             
            getPath("InputSummaryJavadocTestForbiddenFragmentsFoo.java"), expected);          
}                                                                                             
```

Input file:
```java
/*
SummaryJavadoc
violateExecutionOnNonTightHtml = true
forbiddenSummaryFragments = ^Returns the customer ID. This method returns.$
period = .

*/

package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

// ok
public class InputSummaryJavadocTestForbiddenFragmentsFoo {

    // violation 2 lines below
    /**
     * {@summary Returns the customer ID.
     *  This method returns. }
     */
    public void foo() {
    }
}
```

With your changes:
```
Violation lines for /home/vyom/IdeaProjects/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocTestForbiddenFragmentsFoo.java differ.
missing (1): 16
---
expected   : [16]
but was    : []
```

This is both a good and bad regression. 
The above-mentioned is the bad part, here is the good part: (^Returns the customer ID. This method returns. $, **notice the whitespace after `.` and before `$`)**

```java
/*
SummaryJavadoc
violateExecutionOnNonTightHtml = true
forbiddenSummaryFragments = ^Returns the customer ID. This method returns. $
period = .

*/

package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

// ok
public class InputSummaryJavadocTestForbiddenFragmentsFoo {

    // ok, no violation here with the original code, IMO there should be a violation.
    /**
     * {@summary Returns the customer ID.
     *  This method returns. }
     */
    public void foo() {
    }
}
```
with your changes:
```java
/*
SummaryJavadoc
violateExecutionOnNonTightHtml = true
forbiddenSummaryFragments = ^Returns the customer ID. This method returns. $
period = .

*/

package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

// ok
public class InputSummaryJavadocTestForbiddenFragmentsFoo {

    // violation 2 lines below
    /**
     * {@summary Returns the customer ID.
     *  This method returns. }
     */
    public void foo() {
    }
}
```

@romani @rdiachenko Please share your opinion about the above mentioned test case (the one with extra space.)

**Review by Vyom-Yadav — COMMENTED** [review]
Items:
Main item: https://github.com/checkstyle/checkstyle/pull/13311#issuecomment-1616422767

File: config/pitest-suppressions/pitest-javadoc-suppressions.xml (hunk 1 above)
Comment (Kevin222004) [reply config/pitest-suppressions/pitest-javadoc-suppressions.xml:864]: It was their to avoid conflicts

Comment (Vyom-Yadav) [reply config/pitest-suppressions/pitest-javadoc-suppressions.xml:864]: I am ok if you want to remove these at the end of the mutation killing. Resolved.

**Comment by romani** [comment]
@Kevin222004 , please add Inputs from @Vyom-Yadav  to our code base.

>Please share your opinion about the above mentioned test case (the one with extra space.)

It should be a violation. As with inline tags we consider all until `}`. Period property is not in use.

**Comment by Kevin222004** [comment]
@romani done test added

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+forbiddenSummaryFragments = ^Returns the customer ID. This method returns.$
+period = .
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
+
+public class InputSummaryJavadoc1 {
+
+    // violation 2 lines below 'd'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java:13]: >`d`

???

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java:13]: And please create new issue on this Check to not do trim. User owns all symbols.
We do not make any opinion why trailing spaces are required. User should match on whole content.

Let's presume that user wants to violate trailing spaces in inline tags, he should be able to do with simple regexp

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Review by romani — APPROVED** [review]
ok to merge

**Review by Vyom-Yadav — APPROVED** [review]
LGTM! @Kevin222004 Please create an issue for the other case I mentioned in the PR. @romani CircleCI failed due to a connection reset. Please re-run.

**Comment by romani** [comment]
@Kevin222004 , please review what is not done

**Comment by romani** [comment]
@Vyom-Yadav , I think you mentioned only one case and we opened issue for it https://github.com/checkstyle/checkstyle/issues/13347

**Comment by Kevin222004** [comment]
Yes I messaged him for same but not get replied yet

**Comment by Vyom-Yadav** [comment]
https://github.com/checkstyle/checkstyle/issues/13347#issuecomment-1620198157

**Pull request merged**